src/main/java/org/openstreetmap/atlas/geography/atlas/items/complex/aoi/ComplexAOIFinder.java

```diff
@@ -0,0 +1,64 @@
+package org.openstreetmap.atlas.geography.atlas.items.complex.aoi;
+
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.openstreetmap.atlas.geography.atlas.Atlas;
+import org.openstreetmap.atlas.geography.atlas.items.complex.Finder;
+import org.openstreetmap.atlas.tags.filters.TaggableFilter;
+import org.openstreetmap.atlas.utilities.collections.Iterables;
+import org.openstreetmap.atlas.utilities.collections.MultiIterable;
+
+/**
+ * {@link ComplexAOI} finder.
+ *
+ * @author sayas01
+ */
+public class ComplexAOIFinder implements Finder<ComplexAOI>
+{
+    /**
+     * Finds all relations and areas that are candidates for {@link ComplexAOI} and converts them
+     * into {@link ComplexAOI}.
+     *
+     * @param atlas
+     *            The {@link Atlas} to browse.
+     * @return {@link Iterables} of {@link ComplexAOI}.
+     */
+    @Override
+    public Iterable<ComplexAOI> find(final Atlas atlas)
+    {
+        final Iterable<ComplexAOI> iterableOfComplexAOIRelations = Iterables
+                .asIterable(StreamSupport.stream(atlas.relations().spliterator(), false)
+                        .map(ComplexAOI::getComplexAOI).filter(Optional::isPresent)
+                        .map(Optional::get).collect(Collectors.toList()));
```

Comment:
If you collect to List here, shouldn't it already be iterable? My guess is that it does not need the Iterables.asIterable wrapper?

Reply:
You are right! I have removed the wrapper.